### PR TITLE
Fix basic functionality with type `Object` and new analyzer

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Found by WingFuzz team with the help of [Griffin](https://dl.acm.org/doi/abs/10.1145/3551349.3560431) tool.

### CODE REVIEW

File: tests/queries/0_stateless/02553_type_object_analyzer.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+SET allow_experimental_object_type = 1;
+SET allow_experimental_analyzer = 1;
+
+DROP TABLE IF EXISTS t_json_analyzer;
+CREATE TABLE t_json_analyzer (a JSON) ENGINE = Memory;
+INSERT INTO t_json_analyzer VALUES ('{"id": 2, "obj": {"k2": {"k3": "str", "k4": [{"k6": 55}]}, "some": 42}, "s": "bar"}');
+
+SELECT any(a) AS data FROM t_json_analyzer FORMAT JSONEachRow;
+SELECT flattenTuple(a) AS data FROM t_json_analyzer FORMAT JSONEachRow;
+SELECT  a.id, a.obj.* FROM t_json_analyzer FORMAT JSONEachRow;
```
Comment (CurtizJ) [inline tests/queries/0_stateless/02553_type_object_analyzer.sql:11]: This is pretty cool, BTW.

File: src/Analyzer/Passes/QueryAnalysisPass.cpp [hunk 2]
```diff
@@ -5391,7 +5391,12 @@ void QueryAnalyzer::initializeTableExpressionColumns(const QueryTreeNodePtr & ta
     {
         const auto & storage_snapshot = table_node ? table_node->getStorageSnapshot() : table_function_node->getStorageSnapshot();
 
-        auto column_names_and_types = storage_snapshot->getColumns(GetColumnsOptions(GetColumnsOptions::All).withSubcolumns().withVirtuals());
+        auto column_names_and_types = storage_snapshot->getColumns(
```
Comment (kitaisreal) [inline src/Analyzer/Passes/QueryAnalysisPass.cpp:5394]: Please modify it like it is done here https://github.com/ClickHouse/ClickHouse/pull/45461/files#diff-377c0cb9048fe3fe24b9501a4d6fdadde00a04c75a162f705de66cadeb654d9cR5630. We should not use `withSubcolumns` is storage does not support it.

**Review by kitaisreal — CHANGES_REQUESTED** [review]

File: src/Analyzer/Passes/QueryAnalysisPass.cpp (hunk 2 above)
Comment (CurtizJ) [reply src/Analyzer/Passes/QueryAnalysisPass.cpp:5394]: Maybe just close this PR then and move tests to your PR?

Comment (kitaisreal) [reply src/Analyzer/Passes/QueryAnalysisPass.cpp:5394]: Let's merge this one, I will just use them after merge.

**Comment by kitaisreal** [comment]
@CurtizJ fuzzer found a lot of crashes. It seems that they are related to Object type, because fuzzer is disabled for analyzer.

**Review by kitaisreal — APPROVED** [review]

**Comment by CurtizJ** [comment]
Test failures are unrelated.

**Pull request merged**